src/_zkapauthorizer/model.py

```diff
@@ -422,14 +426,22 @@ def insert_unblinded_tokens_for_voucher(
                 token_count_increase,
                 sequestered_count_increase,
                 self.now(),
-                voucher.decode("ascii"),
+                voucher_text,
             ),
         )
         if cursor.rowcount == 0:
             raise ValueError(
                 "Cannot insert tokens for unknown voucher; add voucher first"
             )
 
+        cursor.execute(
+            """
+            SELECT [counter] FROM [vouchers] WHERE [number] = ?
+            """,
+            (voucher_text,),
+        )
+        (new_counter,) = cursor.fetchone()
```

Comment:
I guess there isn't a way to get the value set in the update statement two lines up. :upside_down_face: